src/core/Akka.TestKit/TestKitBase_ExpectAllWithPredicates.cs

```diff
@@ -0,0 +1,155 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Akka.TestKit.Internal;
+
+namespace Akka.TestKit;
+
+public abstract partial class TestKitBase
+{
+    public IReadOnlyCollection<object> ExpectMsgAllOfMatchingPredicates(params PredicateInfo[] predicates)
+    {
+        using var cts = new CancellationTokenSource(RemainingOrDefault);
+        return ExpectMsgAllOfMatchingPredicates(predicates, cts.Token);
+    }
+
+    public IReadOnlyCollection<object> ExpectMsgAllOfMatchingPredicates(IReadOnlyCollection<PredicateInfo> predicates,
+        CancellationToken cancellationToken)
+    {
+        return ExpectMsgAllOfMatchingPredicatesAsync(predicates, cancellationToken)
+            .ToListAsync(cancellationToken)
+            .ConfigureAwait(false).GetAwaiter().GetResult();
+    }
+
+    public async IAsyncEnumerable<object> ExpectMsgAllOfMatchingPredicatesAsync(
+        IReadOnlyCollection<PredicateInfo> predicates,
+        [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        var enumerable = InternalExpectMsgAllOfMatchingPredicatesAsync(new TimeSpan(0, 0, 60), predicates,
+                cancellationToken: cancellationToken)
+            .ConfigureAwait(false).WithCancellation(cancellationToken);
+        await foreach (var item in enumerable)
+        {
+            yield return item;
+        }
+    }
+
+    public IReadOnlyCollection<object> ExpectMsgAllOfMatchingPredicates(TimeSpan max, params PredicateInfo[] predicates)
+    {
+        using var cts = new CancellationTokenSource(RemainingOrDefault);
+        return ExpectMsgAllOfMatchingPredicates(max, predicates, cts.Token);
+    }
+
+    public IReadOnlyCollection<object> ExpectMsgAllOfMatchingPredicates(TimeSpan max,
+        IReadOnlyCollection<PredicateInfo> predicates, CancellationToken cancellationToken)
+    {
+        return ExpectMsgAllOfMatchingPredicatesAsync(max, predicates, cancellationToken)
+            .ToListAsync(cancellationToken)
+            .ConfigureAwait(false).GetAwaiter().GetResult();
+    }
+
+    public async IAsyncEnumerable<object> ExpectMsgAllOfMatchingPredicatesAsync(
+        TimeSpan max,
+        IReadOnlyCollection<PredicateInfo> predicates,
+        [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        max.EnsureIsPositiveFinite("max");
+        var enumerable = InternalExpectMsgAllOfAsync(Dilated(max), predicates, cancellationToken: cancellationToken)
+            .ConfigureAwait(false).WithCancellation(cancellationToken);
+        await foreach (var item in enumerable)
```

Comment:
Same as above

Reply:
More [cargo garbage](https://github.com/malclear/akka.net/blob/feature/addPredicateSetTesting/src/core/Akka.TestKit/TestKitBase_Expect.cs#L654)?

Reply:
Yes, we should return the result instead.

Reply:
Actually, nevermind, this code is good as is, its not possible to return an IAsyncEnumerable with .ConfigureAwait()